### PR TITLE
Make into an Ember CLI Addon.

### PR DESCRIPTION
**Pull request by rwjblue**

### CODE REVIEW

**Comment by ryanflorence** [comment]
Uhhhhhhhhh ... I have not been following, what on earth is going on?

**Comment by rwjblue** [comment]
Ember CLI looks for a keyword of `ember-addon` in the `package.json`.  Then if it finds the keyword it will look at either the `main` or `ember-addon-main` property to instantiate an instance of an `Addon`.  The addon just needs to have a few methods defined to work.  `included` gets called from an EmberCLI projects `Brocfile.js` when each addon is included into the app, and `treeFor` is called when Ember CLI processes its `vendor` trees (these are essentially `bower` or `npm` deps).  In this case the `app.import` line just tells Ember CLI to append the ic-ajax `main.js` to the generated `assets/vendor.js` file, and the vendor tree is merged in with the apps `vendor/` tree.

I am attempting to completely obliterate https://github.com/rjackson/ember-cli-ic-ajax from the face of the planet.  I greatly dislike superfluous packages, and that one is really just repackaging `ic-ajax` for consumption with Ember CLI.

**Comment by rwjblue** [comment]
A few articles for reference:

http://reefpoints.dockyard.com/2014/06/24/introducing_ember_cli_addons.html
http://hashrocket.com/blog/posts/building-ember-addons

**Comment by ryanflorence** [comment]
> I greatly dislike superfluous packages, and that one is really just repackaging ic-ajax for consumption with Ember CLI

There are 5 module formats included in ic-ajax, certainly ember-cli can consume one of them w/o writing 70 lines of code?

**Comment by rwjblue** [comment]
I am happy to write this in fewer lines of code if that would resolve your concerns.

 I would prefer not to, but if this is unacceptable to you, I can continue to maintain the wrapper package.

**Comment by stefanpenner** [comment]
I'm not quite sure why ember-cli-addon is needed for pulling in ic-ajax. Although they can inject code, this doesn't seem like the path we should take.

In-general the original idea of these addons was to provide extra functionality to the CLI itself, generators/commands/pre-processors etc, and _could_ injected client side code but in nearly all cases I don't feel it should be used to inject code.

Cases where I think it should inject code:
- test frameworks
- live-reload
- something like socket.io that has firm client-side and serverside dependencies

I do believe we are yet finished with the client-side addon consumption and ember-cli-addon's are seen as a solution for today, I apologies for being busy and not completing my efforts in this area yet.

Just a tid-bit on this, I have been chatting with @guybedford for proper JSPM integration which utilizes existing distribution methods, but normalizes the consumed packages for ES6 consumption. I suspect this will be the way forward.

As for ic-ajax, we are actually nearly able to support importing without any boiler-plate and without named-amd. I need to push that project to completion, I will try to do so this weekend.

**Comment by ryanflorence** [comment]
> As for ic-ajax, we are actually nearly able to support importing without any boiler-plate and without named-amd. I need to push that project to completion, I will try to do so this weekend.

Sounds lovely :)
